tests/integration_tests/instances.py

```diff
@@ -43,21 +52,37 @@ def emit_settings_to_log(self) -> None:
     def destroy(self):
         self.instance.delete()
 
-    def execute(self, command):
-        return self.instance.execute(command)
+    def execute(self, command, *, use_sudo=None) -> Result:
+        if use_sudo is None:
+            use_sudo = self.use_sudo
+        return self.instance.execute(command, use_sudo=use_sudo)
 
-    def pull_file(self, remote_file, local_file):
-        self.instance.pull_file(remote_file, local_file)
+    def pull_file(self, remote_path, local_path):
+        # First copy to a temporary directory because of permissions issues
+        tmp_path = _get_tmp_path()
+        self.instance.execute('cp {} {}'.format(remote_path, tmp_path))
+        self.instance.pull_file(tmp_path, local_path)
```

Comment:
You're not cleaning up the temp file on the other side.

And it seems like if 'pull_file' is broken, then you should fix it in pycloudlib rather than working around with a temp file.  Now, instead of just using 'pull_file', you'll have:

 * a race condition or failure path based on your _get_tmp_path() colliding with existing files
 * an extra process 'cp'
 * an extra process for 'rm'

Reply:
Changed my tmp file to use a uuid instead so there's no possibility of collisions. I don't think I care about cleaning up the remote end. These are relatively short-lived tests, so I see no harm and it saves us an ssh call.